mapillary_tools/download_blurred.py

```diff
@@ -0,0 +1,413 @@
+             user_name,
+             number_threads=4):
+    """
+    The main function to spin up the downloads threads.
+
+    :param client_id: CLIENT_ID required for requests
+    :type client_id: string
+
+    :param organization_keys: list of orgs to filter by
+    :type  organization_keys: List<string>
+
+    :param start_time: start of time range to use for requests
+    :type  start_time: None or string (YYYY-MM-DD)
+
+    :param end_time: end of time range to use for requests
+    :type  end_time: None or string (YYYY-MM-DD)
+
+    :param output_folder: destination for the files
+    :type output_folder: string
+
+    :param user_name: user name of the requesting user
+    :type  user_name: string
+    """
+    signal.signal(signal.SIGTERM, service_shutdown)
+    signal.signal(signal.SIGINT, service_shutdown)
+
+    headers = get_headers_for_username(user_name)
+
+    start_time_b64 = '' if not start_time else start_time
+    end_time_hash = '' if not end_time else end_time
+    download_id = b','.join([client_id,
+                             organization_keys,
+                             start_time_b64,
+                             end_time_hash,
+                             output_folder,
+                             user_name,
+                             ])
+    hash_object = hashlib.md5(download_id)
+    download_id = str(hash_object.hexdigest())
+
+    # create directories for saving
+    create_dirs(output_folder)
+    save_path = download_id
+
+    # create most of the bookkeeping files
+    all_file = '{}_all.txt'.format(save_path)
+    all_file_exists = os.path.isfile(all_file)
+    all_file_obj = open(
+        all_file, 'r') if all_file_exists is True else open(all_file, 'a+')
+
+    done_file = '{}_done.txt'.format(save_path)
+    done_file_exists = os.path.isfile(done_file)
+
+    err_file = '{}_err.txt'.format(save_path)
+    err_file_exists = os.path.isfile(err_file)
+
+    # fetch image keys and store basic state for downloads from bookkeeping files
+    image_keys = []
+    if all_file_exists:
+        all_keys = all_file_obj.read().split('\n')
+
+        done_file_obj = open(
+            done_file, 'r+') if done_file_exists is True else open(done_file, 'r+')
+
+        err_file_obj = open(
+            err_file, 'r') if err_file_exists is True else open(err_file, 'r+')
+
+        done_keys = done_file_obj.read().split('\n')
+        err_keys = err_file_obj.read().split('\n')
+        set_all = set(all_keys)
+        set_done = set(done_keys)
+        set_err = set(err_keys)
+
+        diff = list(set_all.difference(set_done).difference(set_err))
+        print("Download {} continued: {}/{}".format(download_id, len(diff), len(all_keys)))
+
+        image_keys = diff
+        done_file_obj.close()
+        err_file_obj.close()
+    else:
+        headers = get_headers_for_username("knikel")
+        image_keys = query_search_api(
+            headers,
+            client_id=uploader,
```

Comment:
you likely going to remove this for release, but just a note .CLIENT_ID is missing

Reply:
Yes, thank you!